src/lapack/backends/cusolver/cusolver_batch.cpp

```diff
@@ -79,251 +100,642 @@ void getri_batch(sycl::queue &queue, std::int64_t n, sycl::buffer<std::complex<d
+                                      nrhs, a_ + stride_a * i, lda, ipiv_ + stride_ipiv * i,
+                                      b_ + stride_b * i, ldb, nullptr);
+            }
+            CUSOLVER_SYNC(err, handle)
+        });
+    });
 }
-void potrs_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, std::int64_t n, std::int64_t nrhs,
-                 sycl::buffer<double> &a, std::int64_t lda, std::int64_t stride_a,
-                 sycl::buffer<double> &b, std::int64_t ldb, std::int64_t stride_b,
-                 std::int64_t batch_size, sycl::buffer<double> &scratchpad,
-                 std::int64_t scratchpad_size) {
-    throw unimplemented("lapack", "potrs_batch");
+
+#define GETRS_STRIDED_BATCH_LAUNCHER(TYPE, CUSOLVER_ROUTINE)                                      \
+    void getrs_batch(sycl::queue &queue, oneapi::mkl::transpose trans, std::int64_t n,            \
+                     std::int64_t nrhs, sycl::buffer<TYPE> &a, std::int64_t lda,                  \
+                     std::int64_t stride_a, sycl::buffer<std::int64_t> &ipiv,                     \
+                     std::int64_t stride_ipiv, sycl::buffer<TYPE> &b, std::int64_t ldb,           \
+                     std::int64_t stride_b, std::int64_t batch_size,                              \
+                     sycl::buffer<TYPE> &scratchpad, std::int64_t scratchpad_size) {              \
+        return getrs_batch(#CUSOLVER_ROUTINE, CUSOLVER_ROUTINE, queue, trans, n, nrhs, a, lda,    \
+                           stride_a, ipiv, stride_ipiv, b, ldb, stride_b, batch_size, scratchpad, \
+                           scratchpad_size);                                                      \
+    }
+
+GETRS_STRIDED_BATCH_LAUNCHER(float, cusolverDnSgetrs)
+GETRS_STRIDED_BATCH_LAUNCHER(double, cusolverDnDgetrs)
+GETRS_STRIDED_BATCH_LAUNCHER(std::complex<float>, cusolverDnCgetrs)
+GETRS_STRIDED_BATCH_LAUNCHER(std::complex<double>, cusolverDnZgetrs)
+
+#undef GETRS_STRIDED_BATCH_LAUNCHER
+
+template <typename Func, typename T>
+inline void getrf_batch(const char *func_name, Func func, sycl::queue &queue, std::int64_t m,
+                        std::int64_t n, sycl::buffer<T> &a, std::int64_t lda, std::int64_t stride_a,
+                        sycl::buffer<std::int64_t> &ipiv, std::int64_t stride_ipiv,
+                        std::int64_t batch_size, sycl::buffer<T> &scratchpad,
+                        std::int64_t scratchpad_size) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+
+    overflow_check(m, n, lda, stride_a, stride_ipiv, batch_size, scratchpad_size);
+
+    // cuSolver legacy api does not accept 64-bit ints.
+    // To get around the limitation.
+    // Create new buffer with 32-bit ints then copy over results
+    std::uint64_t ipiv_size = stride_ipiv * batch_size;
+    sycl::buffer<int, 1> ipiv32(sycl::range<1>{ ipiv_size });
+    sycl::buffer<int> devInfo{ batch_size };
```

Comment:
`sycl::buffer<int>` is equivalent to `sycl::buffer<int, 1>`

Reply:
Thanks for catching this, I have removed the dim value.